src/main/scala/is/hail/utils/SummaryCombiner.scala

```diff
@@ -39,28 +41,49 @@ class SummaryCombiner[T](f: Iterable[T] => Int) extends Serializable {
     this
   }
 
-  def merge(data: (Variant, (Annotation, Iterable[T]))): SummaryCombiner[T] = {
+  def merge(rv: RegionValue): SummaryCombiner = {
+    require(rowType.fieldType(1).isInstanceOf[TVariant])
+    val vType = rowType.fieldType(1).asInstanceOf[TVariant]
+    val r = rv.region
+    val vOffset = rowType.loadField(rv, 1)
+    val t = vType.representation
+    val altsType = t.fieldType(3).asInstanceOf[TArray]
+
     nVariants += 1
 
-    val v = data._1
-    contigs += v.contig
-    if (v.nAlleles > 2)
+    val contig = TString.loadString(r, t.loadField(r, vOffset, 0))
+    contigs += contig
+
+    val altsOffset = t.loadField(r, vOffset, 3)
+    val nAlts = altsType.loadLength(r, altsOffset)
+    val nAlleles = nAlts + 1
+    if (nAlleles > 2)
       multiallelics += 1
-    if (v.nAlleles > maxAlleles)
-      maxAlleles = v.nAlleles
+    if (nAlleles > maxAlleles)
+      maxAlleles = nAlleles
 
-    v.altAlleles.foreach { aa =>
-      aa.altAlleleType match {
+    var i = 0
+    while (i < nAlts) {
+      val altOffset = altsType.loadElement(r, altsOffset, nAlts, i)
+      val altAllele = AltAllele.fromRegionValue(r, altOffset)
+      altAllele.altAlleleType match {
         case AltAlleleType.SNP => snps += 1
         case AltAlleleType.MNP => mnps += 1
         case AltAlleleType.Insertion => insertions += 1
         case AltAlleleType.Deletion => deletions += 1
         case AltAlleleType.Complex => complex += 1
         case AltAlleleType.Star => star += 1
       }
+      i += 1
     }
 
-    nCalled += f(data._2._2)
+    view.setRegion(rv)
```

Comment:
You should create the view in a mapPartitions call and pass it in to this function.  Same for the variant view.

But you're using an aggregate.  How to turn that into a mapPartitions?  Something like:

```
  rdd2.mapPartitions { it =>
    val comb = new SummaryCombiner
    val view = ...
    it.map { rv => comb.merge(rv, view) }
    Iterator(comb)
  }.aggregate(_.merge(_), _.merge(_))
```

Reply:
Might be nice to have:

```scala
def rdd2Aggregate[T](setup: () => T)(seqOp: (T, RV) => (), combOp (T, T) => (T)): T =
  rdd2.mapPartitions { it =>
    val t = setup()
    it.map { rv => seqOp(t, rv) }
    Iterator(t)
  }.aggregate(combOp, combOp)
```

so we can do

```scala
rdd2.rdd2Aggregate(() => (new SummaryCombiner(), ...))(
  { case (c, v) => c.merge(v, view) }, _.merge(_))
```